File: php/test/Behavioral/StateTest.php
<?

require_once 'src/Behavioral/State/DvdStateContext.php';

class Behavioral_StateTest extends PHPUnit_Framework_TestCase
{
   public function test()
   {
		$stateContext = new DvdStateContext();
		$this->assertEquals($stateContext->getName( "Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies"), 'Sponge*Bob*Squarepants*-*Nautical*Nonsense*and*Sponge*Buddies', 'first state change');
		$this->assertEquals($stateContext->getName( "Jay and Silent Bob Strike Back" ) , 'Jay*and*Silent*Bob*Strike*Back', 'second state change');
		$this->assertEquals($stateContext->getName( "Buffy The Vampire Slayer Season 2"), 'Buffy!The!Vampire!Slayer!Season!2', 'third state change');
		$this->assertEquals($stateContext->getName( "The Sopranos Season 2"), 'The*Sopranos*Season*2', 'fourth state change');
	}
}

File: clojure/test/patterns/behavioral/state_test.clj
(ns patterns.behavioral.state-test
  (:require 
		[clojure.test :refer :all]
      [patterns.behavioral.state :refer :all]
	))

(deftest tst
	(let [munger (name-munger)]
		(testing "First show name"
			(is	(= "Sponge*Bob*Squarepants*-*Nautical*Nonsense*and*Sponge*Buddies" 
					(munge-name munger "Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies"))))
		(testing "Second show name"
			(is (= "Jay*and*Silent*Bob*Strike*Back" 
				 (munge-name munger "Jay and Silent Bob Strike Back"))))

		(testing "Third show name"
			(is (= "Buffy!The!Vampire!Slayer!Season!2" 
				 (munge-name munger "Buffy The Vampire Slayer Season 2"))))

(comment
		(testing "Fourth show name"
			(is (= "The*Sopranos*Season*2" 
				 (munge-name name-munger "The Sopranos Season 2"))))
)
))
